### PR TITLE
Added ResNext models

### PR DESCRIPTION
**Pull request by jaketae**
Added ResNext models. It seems like the implementation of ResNext models was dropped, despite them being available options in `WEIGHT_HASHES`. The implementation can simply be achieved by using `stack3`, which is also a function that was declared and not used.

Addresses tensorflow/models#1248, tensorflow/models#6764.

### CODE REVIEW

**Review by tanzhenyu — DISMISSED** [review]
This looks good to me. I don't know why we dropped them in the first place (looks like it's missed?)
Please wait for Francois' LGTM.

**Comment by gbaned** [comment]
@jaketae Can you please address Ubuntu Sanity errors? Thanks!

**Comment by jaketae** [comment]
@tanzhenyu  @gbaned Please correct me if I'm wrong, but to my knowledge the CI failures seem to be happening due to things unrelated to this PR. I'm by no means familiar with CI's automated testing, but here's an [example log](https://source.cloud.google.com/results/invocations/da266721-5903-4975-a630-8f3a80d978a9/targets/%2F%2Ftensorflow%2Fcompiler%2Ftests:sort_ops_test_cpu/log) for reference:

```
if (!operand_cst->isVariableLength()) {
           ~~~~~~~~~~~  ^
tensorflow/compiler/mlir/xla/operator_writer_gen.cc:111:33: error: no member named 'getNumVariableLengthOperands' in 'mlir::tblgen::Operator'; did you mean 'getNumVariadicOperands'?
  if (op.getNumOperands() == op.getNumVariableLengthOperands()) {
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                getNumVariadicOperands
external/llvm-project/mlir/include/mlir/TableGen/Operator.h:145:12: note: 'getNumVariadicOperands' declared here
  unsigned getNumVariadicOperands() const;
```

Could you give me some pointers on where to go from here? Thank you in advance.

**Comment by byronyi** [comment]
@jaketae See b7c1d24656c61b39375635644dd3f168e9a811cb which fixes https://github.com/tensorflow/tensorflow/commit/f5ee7354284987cb0449c2c4be5a7b81717babaf#commitcomment-38456646.

cc @d0k

**Comment by jaketae** [comment]
@byronyi Got it. So since a fix was pushed out, a quick merge and a CI rerun should address the problem? I'll try it today and get back to you.

**Comment by byronyi** [comment]
> @byronyi Got it. So since a fix was pushed out, a quick merge and a CI rerun should address the problem? I'll try it today and get back to you.

I think you don't even need to rebase/merge any commits for this PR. Simply kicking off CI again will do.

**Comment by tanzhenyu** [comment]
Thanks Jake. I realized that 1) this needs API proto generation, which you probably wouldn't be able to do so; 2) this needs basic unit test, 3) there's a bug on line of code to `load_weights`, you need a comma in front of `by_name`.

So I'm gonna create a commit instead and acknowledge your PR inside the commit.

**Comment by jaketae** [comment]
@tanzhenyu, thank you for the review. Yes, the API proto generation part is well beyond my reach, so I agree that this is the best way to go. Looking forward to your PR!

**Comment by gbaned** [comment]
@tanzhenyu Any update on this PR? Please. Thanks!

File: tensorflow/python/keras/applications/resnet.py
Comment (Ir1d) [inline tensorflow/python/keras/applications/resnet.py:223]: Should this be 
```suggestion
    model.load_weights(weights_path, by_name=by_name)
```

Comment (jaketae) [reply tensorflow/python/keras/applications/resnet.py]: Good catch, thanks! I'm still waiting on @tanzhenyu though, we'll see how this goes.

**Comment by gbaned** [comment]
@tanzhenyu Any update on this PR? Please. Thanks!

**Review by tanzhenyu — APPROVED** [review]

**Comment by ptran1203** [comment]
There is an error when input_shape is (None, None, 3)

image

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

I'm taking the review over from Zhenyu.

Please add unit tests for this change: the new applications should pass all of the standard Keras Applications tests (see the corresponding test file).

**Comment by gbaned** [comment]
@jaketae  Can you please check @fchollet's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@jaketae  Any update on this PR? Please. Thanks!

**Comment by jaketae** [comment]
> @jaketae Any update on this PR? Please. Thanks!

My apologies, I've been caught up lately with some other things. I'll try to make changes to the PR as needed throughout this week (and potentially beyond, depending on how much progress I make). Thanks for the friendly reminders!

**Comment by fchollet** [comment]
The unit tests could be as simple as just adding the next models to the list of covered models in `keras/applications/applications_test.py`.

**Comment by gbaned** [comment]
@jaketae Can you please check @fchollet's comments and keep us posted ? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by jaketae** [comment]
I just added the two models in the unittest, but I'm unsure about some details. I hope to get this PR finalized by this weekend, should any additional work and debugging be necessary (waiting for the CI at the moment).

**Review by fchollet — APPROVED** [review]
LGTM, thanks for the contribution! 👍

**Comment by jaketae** [comment]
@fchollet Thank you for the review! Before I get too excited though, can you give me some pointers as to how to debug the backward compatibility issues the CI is complaining about? I'm assuming it has to be attended to, but I'm not entirely sure how to interpret the log. Thanks in advance!

**Comment by fchollet** [comment]
Hi @jaketae,

Don't worry about the CI check about API files. We'll take care of it.

I was looking at merging this earlier and there's a problem we need to fix. First of all, please add the new models to the test `keras/applications/applications_load_weight_test.py`, like this:

```
ARG_TO_MODEL = {
    'resnet': (resnet, [resnet.ResNet50, resnet.ResNet101, resnet.ResNet152,
                                  resnet.ResNeXt50, resnet.ResNeXt101]),
   ...
```

When you do this, you will see that the test fails because the weights are not loaded correctly. The model's predictions are incorrect after loaded the pretrained weights.

We need to fix this. In the process of fixing it we need to get rid of the `by_name = True` flag. Instead of we should be loading weights normally, from a checkpoint that is specific to each new model. So you'll have to prepare two new checkpoint files, make sure that the test `applications_load_weight_test.py` passes with the files, then upload the checkpoints to GitHub or a cloud bucket. Then we'll take it over from there.

**Comment by gbaned** [comment]
@jaketae Can you please check @fchollet's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@jaketae  Any update on this PR? Please. Thanks!

**Comment by realeve** [comment]
> Hi @jaketae,
> 
> Don't worry about the CI check about API files. We'll take care of it.
> 
> I was looking at merging this earlier and there's a problem we need to fix. First of all, please add the new models to the test `keras/applications/applications_load_weight_test.py`, like this:
> 
> ```
> ARG_TO_MODEL = {
>     'resnet': (resnet, [resnet.ResNet50, resnet.ResNet101, resnet.ResNet152,
>                                   resnet.ResNeXt50, resnet.ResNeXt101]),
>    ...
> ```
> 
> When you do this, you will see that the test fails because the weights are not loaded correctly. The model's predictions are incorrect after loading the pretrained weights.
> 
> We need to fix this. In the process of fixing it we need to get rid of the `by_name = True` flag. Instead of that we should be loading weights normally, from a checkpoint that is specific to each new model. So you'll have to prepare two new checkpoint files, make sure that the test `applications_load_weight_test.py` passes with the files, then upload the checkpoints to GitHub or a cloud bucket. Then we'll take it over from there.


Is the problem fixed? thanks.

**Comment by gbaned** [comment]
@jaketae  Any update on this PR? Please. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
@jaketae Any update on this PR? Please. Thanks!

**Comment by jaketae** [comment]
Huge apologies for the belated update. I haven't had the time to look into this recently, which is compounded by the fact that I don't have enough resources to run experiments and train the model to optimality. I'd be happy to relay this issue to anyone else who'd be interested. If not, I'll try to find time and the compute to prepare model weights.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 45 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

**Pull request closed without merging**
